pkg/operator/ceph/csi/spec.go

```diff
@@ -556,16 +556,28 @@ func createCSIDriverInfo(clientset kubernetes.Interface, name string) error {
 		},
 	}
 	csidrivers := clientset.StorageV1beta1().CSIDrivers()
-	_, err := csidrivers.Create(csiDriver)
```

Comment:
Instead of the pattern `Get()`, `Create()` if doesn't exist, or `Update()` if it does exist, this is a more common pattern:
- `Create()`
- If already exists, `Update()`. If there are not any changes to the resource, K8s should ignore the update.

This way, if there are any other changes to the CSIDriver spec, we will always pick them up during upgrade, and no need to make specific checks for things like the owner references.

Reply:
I tried to do it but got below error

```
ceph-csi: failed to start Ceph csi drivers. failed to create CSI driver object for "rook-ceph.rbd.csi.ceph.com": csidrivers.storage.k8s.io "rook-ceph.rbd.csi.ceph.com" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update
```

I have to do get and replace the resourceVersion in the current object and update so that any changes to  the CSIDriver spec can take place

Reply:
The first return value for the call to `Create()` is the existing object, which you could get the resourceVersion and apply it on the update, right?

Reply:
Correct I forgot about that object 🤦  let me test it out and update

Reply:
I tried the below code, it failed with the same error ```csidrivers.storage.k8s.io "rook-ceph.rbd.csi.ceph.com" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update```

```go
csidrivers := clientset.StorageV1beta1().CSIDrivers()

	driver, err := csidrivers.Create(csiDriver)
	if err == nil {
		logger.Infof("CSIDriver object created for driver %q", name)
		return nil
	}

	if apierrors.IsAlreadyExists(err) {
		csiDriver.ObjectMeta.ResourceVersion = driver.ObjectMeta.ResourceVersion
		_, err = csidrivers.Update(csiDriver)
		if err == nil {
			logger.Infof("CSIDriver object updated for driver %q", name)
		}
	}

	return err
```

Reply:
Was the `driver.ObjectMeta.ResourceVersion` blank or was it actually set? If this approach still doesn't work, we can go with your original approach.

Reply:
I verified it we are not getting ResourceVersion in the driver object when we create the resource so its not possible to use create--> update approch.